docs/user-guide/Authentication.md

```diff
@@ -4,10 +4,26 @@ RESTler supports token-based and certificate based authentication.
 
 **Token based authentication**
 
-The user must provide a separate program to generate tokens, which implements the authentication method required by the API.  This will be invoked in a separate process by RESTler to obtain and regularly refresh tokens.  When invoked, this program must print metadata about the tokens on the first line, followed by each token and the required token header on a separate line for each application.  For example:
+
+The user has three options for providing token based authentication; Module, Location, and CMD.
+
+**Module**
+
+The user must provide the path to a python module (.py) that implements a function to acquire a token, and the name of the function (default: "acquire_token"). RESTler will import the module and call the function. Additionally, a user can opt to add "data", which will be converted into a dictionary and passed to their function. The function must accept two parameters "data", a Dictionary containing the json payload specified under data, and "log" a method that will write any logs to a network auth text file. 
```

Comment:
>Additionally, a user can opt to add "data", which will be converted into a dictionary and passed to their function. The function must accept two parameters "data", a Dictionary containing the json payload specified under data, and "log" a method that will write any logs to a network auth text file.  [](http://example.com/codeflow?start=214&length=300)

Could you please update this part to be a little more structured, for example:

Additionally, a user can opt to add ```data``` (e.g. with additional authentication-related parameters specific to the service under test) to pass to this function.  The function signature must be as follows:

```
def acquire_token(data, log):
   ...
```
where
- ```data``` is a dictionary containing the json payload specified in the corresponding engine setting (see SettingsFile.md)
- ```log``` is a function that may be used to write logs to a network auth text file that will be saved in the RESTler results directory next to the network logs.

Reply:
Sure, I think your example looks good, will add 

Also making a minor tweak to specify that the function should return a token